New take on a docker setup

danquah opened the pull request:
TODOs
- Update Readme
- General feedback on structure / naming / etc
- Local testing

danquah committed:
Basic docker dev-setup

tripox commented on README.md:
```diff
@@ -1,2 +1,39 @@
+* Changes in `package.json` should invalidate the Docker image cache.
+If it doesn't rebuild, run `docker-compose up --force-rebuild --build`
+* You can run `npm install` inside the container with `docker-compose exec node npm install`
+* Start indexing manually (starts automatically) with `docker-compose exec node run index all`
```
Should be removed

tripox commented on development.env-dist:
```diff
@@ -0,0 +1,11 @@
+GITHUB_TOKEN=my-token-from-http://
```
Should be a real token or nothing. Maybe a comment above.

tripox commented on scripts/index-all-assets.sh:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+docker-compose exec node npm run index all
```
This should be in the `README.md` and in a Makefile.

tripox commented on docker/dev-env-Dockerfile:
```diff
@@ -0,0 +1,49 @@
+    g++ \
+    supervisor \
+    nginx \
+&& rm -rf /var/lib/apt/lists/* # Keeps the image size down
```
`apt-get clean`?

stumpdk committed:
Updated readme with new instructions

stumpdk commented:
@danquah Jeg strander når jeg kører docker-compose up. Jeg får følgende fejl: ``No such file or directory/env: bash`` fra Node-servicen

stumpdk committed:
Added instructions on how to build the Dockerfile manually

danquah committed:
Make elastic-search accesible via dory

danquah committed:
Switch to a gcp hosted dev image

danquah committed:
Bump our elasticsearch version to the version prod uses

danquah committed:
- Made preparations for cloud builder, added a builder image
- Updated readme to reflect that you now have to do your cloning by hand - deleted
  do-clone.sh/clone-projects.sh consequently
- Renamed build-image to make it clear that its strictly for local use

tripox commented on README.md:
```diff
@@ -6,16 +6,14 @@ Docker setup for kbhbilleder.
 1. [Install Docker](https://store.docker.com/search?type=edition&offering=community)
-2. Clone this repository
-3. [Add an .env file to ./kbhbilleder/](https://github.com/CopenhagenCityArchives/kbh-billeder#create-a-env-file-with-environment-variables)
+2. Dory
```
Please drop a link to [the installation instructions](https://github.com/FreedomBen/dory#installation)

danquah committed:
Adding installation instructions for dory
